test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -107,6 +107,61 @@ public void CheckStateForUnstructuredLogWithStringInterpolation()
             Assert.Equal(message.ToString(), state.ToString());
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1602:Enumeration items should be documented", Justification = "enum for test")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "enum for test")]
+        public enum Field
+        {
+            FormattedMessage,
+            State,
+            StateValues,
+        }
+
+        [Theory]
+        [InlineData(Field.State)]
+        [InlineData(Field.StateValues)]
+        [InlineData(Field.FormattedMessage)]
+        public void SettersTest(Field fieldName)
+        {
+            var exportedItems = new List<LogRecord>();
+            var exporterLoggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(exportedItems, fieldName));
+                }));
+
+            var exporterLogger = exporterLoggerFactory.CreateLogger<LogRecordTest>();
+            if (fieldName == Field.State)
+            {
+                exporterLogger.LogInformation($"This should be replaced {0.99}.");
+                exporterLoggerFactory.Dispose();
```

Comment:
if split to separate tests, you can do using loggerFactory approach, which looks a lot cleaner.
Also, for this test, is Dispose even needed before validating the ExportedItems[0]